src/components/BusMarker.jsx

```diff
@@ -25,34 +26,48 @@ const BusMarker = ({
     setActiveStop,
     setNewReportCoordinates,
     connectionData,
+    draggableStopId,
+    setDraggableStopId,
+    movedStopsDispatch,
+    markerReference,
+    setMarkerReference,
+    setResetPositionFunction,
+    mapMode,
   } = useContext(MapContext);
+  const markerRef = useRef(null);
+  const {lat, lon, id, name, number, stopId, stopType} = busStop;
+  const originalCoordinates = useMemo(() => {
+    return [lat, lon];
+  }, [lat, lon]);
+  const [position, setPosition] = useState(originalCoordinates);
+
   const opacity = useMemo(() => {
-    if (connectedStopIds.includes(busStop.id)) {
+    if (connectedStopIds.includes(id)) {
       return visibilityOptions.connected.value.opacityValue;
     }
-    if (approvedStopIds.includes(busStop.id)) {
+    if (approvedStopIds.includes(id)) {
       return visibilityOptions.approved.value.opacityValue;
     }
     return visibilityOptions.unconnected.value.opacityValue;
-  }, [visibilityOptions, busStop.id, connectedStopIds, approvedStopIds]);
+  }, [visibilityOptions, id, connectedStopIds, approvedStopIds]);
 
-  const handleViewModeStopClick = busStop => {
+  const handleViewModeStopClick = () => {
     clickBusStop(busStop);
-    if (connectedStopIds.includes(busStop.id) || approvedStopIds.includes(busStop.id)) {
+    if (connectedStopIds.includes(id) || approvedStopIds.includes(id)) {
       const connection =
-        busStop.stopType === 0
-          ? importedConnections.find(stop => stop.osmStopId === busStop.id)
-          : importedConnections.find(stop => stop.gtfsStopId === busStop.id);
-      if (busStop.stopType === 0) {
+        stopType === 0
```

Comment:
Please use some more descriptive values instead of 0 ;)
I'd like to have some human readable name here :)